src/App.tsx

```diff
@@ -1,29 +1,89 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { PaginationInfo } from './components/PaginationInfo';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items: number[] = getNumbers(1, 42).map(n => n);
+
+const perPageOptions = [3, 5, 10, 20].map((el, i) => {
+  return { id: i, val: el };
+});
+
+interface PageItem {
+  id: number;
+  val: number;
+}
 
 export const App: React.FC = () => {
+  function buildPagesContainer(
+    itms: number[],
+    chunkSize: number,
+  ): PageItem[][] {
+    const contArr: PageItem[][] = [];
+    let globId = 0;
+
+    if (!chunkSize) {
+      return [];
+    }
+
+    for (let i = 0; i < itms.length; i += chunkSize) {
+      const chunk = itms.slice(i, i + chunkSize);
+      const subCont = [];
+
+      for (let i2 = 0; i2 < chunk.length; i2++) {
+        subCont.push({ id: globId, val: chunk[i2] });
+        globId++;
+      }
+
+      contArr.push(subCont);
+    }
+
+    return contArr;
+  }
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(perPageOptions[1].val);
```

Comment:
The initial `currentPage` and `perPage` states could be derived from constants to avoid magic numbers and make it easier to change these defaults later.